app/Http/Controllers/API/Users/SettingsController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Ushahidi\App\Http\Controllers\API\Users;
+
+use Ushahidi\App\Http\Controllers\RESTController;
+use Illuminate\Http\Request;
+
+/**
+ * Ushahidi API User SettingsTags Controller
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application\Controllers
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+class SettingsController extends RESTController
+{
+	protected function getResource()
+	{
+		return 'user_settings';
+	}
+
+	public function index(Request $request)
+	{
+		$params = $this->getRouteParams($request);
+		
+		$this->usecase = $this->usecaseFactory
+			->get($this->getResource(), 'search');
+		
+		if (is_array($params) && array_key_exists('user_id', $params)) {
+			$this->usecase
+				->setFilters($request->query() + [
```

Comment:
I wonder if we could rewrote this slightly and avoid treating user_id as a filter... its really an identifier ie. part of the primary key, not an optional filter